MapboxSceneKit/Routes/PolylineNode.swift

```diff
@@ -0,0 +1,279 @@
+//
+//  LineNode.swift
+//  line-rendering
+//
+//  Created by Jim Martin on 8/2/18.
+//  Copyright © 2018 Jim Martin. All rights reserved.
+//
+
+import Foundation
+import SceneKit
+import Metal
+
+/// PolylineNode generates line geometry through a list of positions, and uses a Metal Shader to give that geometry a set radius in screen-space. Not compatible with iOS simulator builds.
+///
+/// - Parameters:
+///   - positions: The list of SCNVector3 positions. The line is drawn through each position consectutively from 0...n
+///   - startRadius: The width of the initial point of the line. Linearly interpolated from start to end positions.
+///   - endRadius: The width of the final point of the line. Linearly interpolated from start to end positions.
+///   - startColor: The color of the initial point of the line. Linearly interpolated through RGB color space from start to end.
+///   - endColor: The color of the final point of the line. Linearly interpolated through RGB color space from start to end.
+///   - handleGeometryOverlap: !Experimental! Flag indicating whether the line radius should be used to handle overlap/z-fighting per-pixel. Greatly improves visual quality of lines intersecting with other geometry, but at some cost to performance. May create graphical artifacts on some devices.
+
+@objc(MBPolylineNode)
+public class PolylineNode: SCNNode {
+    
+    private var positions : [SCNVector3] = [SCNVector3Zero]
+    private var startRadius : CGFloat = 0.1
+    private var endRadius : CGFloat = 0.5
+    private var startColor : UIColor = UIColor.yellow
+    private var endColor : UIColor = UIColor.white
+    
+    //TODO: It's possible to set this flag per-vertex to further optimize performance.
+    private var handleGeometryOverlap : Bool = false
+    
+    //derived from position/radius
+    private var verts : [SCNVector3]! //components -> quads
+    private var normals : [SCNVector3]! //Appears as 'neighbors' in the vertex input. Used to pass the line segment(direction and magnitude) to each vertex.
+    private var indices: [Int32]!
+    private var uvs : [CGPoint]! //uv per vert, indicating the corner of the quad
+    private var colors : [SCNVector4]! //vertex colors
+    private var lineParams : [CGPoint]! //line radius captured in y value
+    
+    @objc
+    public init( positions: [SCNVector3],
+                      startRadius: CGFloat, endRadius: CGFloat,
```

Comment:
Just out of curiosity, what is the use case for start/end radius for the` PolyLineNode`

Reply:
@camdeardorff For data visualization, it might be useful to show some data with line weight as well as color ( thinking of visuals like this https://en.wikipedia.org/wiki/Charles_Joseph_Minard). In the future, this could even be done per-vertex for more control.